services/ui-src/src/utils/auth/userProvider.test.tsx

```diff
@@ -107,37 +99,82 @@ describe("Test UserProvider", () => {
     expect(screen.getByTestId("testdiv")).toHaveTextContent("User Test");
   });
 
-  test("test logout function fails", async () => {
-    jest.spyOn(window, "alert").mockImplementation(() => {});
-
+  test("test logout function", async () => {
     await act(async () => {
       const logoutButton = screen.getByTestId("logout-button");
       await userEvent.click(logoutButton);
     });
+    expect(location.pathname).toEqual("/");
+  });
 
-    expect(window.alert).toHaveBeenCalledWith("failed!");
+  test("test login with IDM function", async () => {
+    await act(async () => {
+      const loginButton = screen.getByTestId("login-idm-button");
+      await userEvent.click(loginButton);
+    });
+    expect(screen.getByTestId("testdiv")).toHaveTextContent("User Test");
   });
+});
 
-  test("test logout function", async () => {
-    // stash alert and define temporary mock for use by jest-dom
-    const actualAlert = window.alert;
-    window.alert = () => {};
+describe("Test UserProvider with production path", () => {
+  const originalLocationDescriptor: any = Object.getOwnPropertyDescriptor(
+    global,
+    "location"
+  );
+  const federatedSignInSpy = jest.spyOn(Auth, "federatedSignIn");
 
+  afterAll(() => {
+    Object.defineProperty(global, "location", originalLocationDescriptor);
+    federatedSignInSpy.mockRestore();
+  });
+
+  test("test production authenticates with idm when current authenticated user throws an error", async () => {
+    await setWindowOrigin("mdctmcr.cms.gov");
+    await breakCheckAuthState();
     await act(async () => {
-      const logoutButton = screen.getByTestId("logout-button");
-      await userEvent.click(logoutButton);
+      await render(testComponent);
     });
-    expect(location.pathname).toEqual("/");
+    expect(window.location.origin).toContain("mdctmcr.cms.gov");
+    expect(screen.getByTestId("testdiv")).toHaveTextContent("User Test");
+    expect(federatedSignInSpy).toHaveBeenCalledTimes(2);
+  });
+});
+
+describe("Test UserProvider with non-production path", () => {
+  const originalLocationDescriptor: any = Object.getOwnPropertyDescriptor(
+    global,
+    "location"
+  );
+
+  afterAll(() => {
+    Object.defineProperty(global, "location", originalLocationDescriptor);
+  });
 
-    // restore actual alert method
-    window.alert = actualAlert;
+  test("Non-production error state correctly sets showLocalLogins", async () => {
+    await setWindowOrigin("wherever");
+    await breakCheckAuthState();
+    await act(async () => {
+      await render(testComponent);
+    });
+    expect(window.location.origin).toContain("wherever");
+    const showLocalLogins = screen.getByTestId("show-local-logins");
+    expect(showLocalLogins).toHaveTextContent("showLocalLogins is true");
```

Comment:
This test got that pesky `userProvider` line covered that we were wondering about.